src/gt4py/next/program_processors/runners/dace_fieldview/transformations/k_blocking.py

```diff
@@ -0,0 +1,437 @@
+                continue
+
+        # Handle the Map exits
+        #  This is simple reconnecting, there would be possibilities for improvements
+        #  but we do not use them for now.
+        for out_edge in list(graph.in_edges(outer_exit)):
+            edge_conn = out_edge.dst_conn[3:]
+            dace_helpers.redirect_edge(
+                state=graph,
+                edge=out_edge,
+                new_dst=inner_exit,
+                new_dst_conn="IN_" + edge_conn,
+            )
+            graph.add_edge(
+                inner_exit,
+                "OUT_" + edge_conn,
+                outer_exit,
+                out_edge.dst_conn,
+                copy.deepcopy(out_edge.data),
+            )
+            inner_exit.add_in_connector("IN_" + edge_conn)
+            inner_exit.add_out_connector("OUT_" + edge_conn)
+
+        # TODO(phimuell): Use a less expensive method.
+        dace.sdfg.propagation.propagate_memlets_state(sdfg, graph)
+
+    def partition_map_output(
+        self,
+        map_entry: dace_nodes.MapEntry,
+        block_param: str,
+        state: SDFGState,
+        sdfg: SDFG,
+    ) -> tuple[set[dace_nodes.Node], set[dace_nodes.Node]] | None:
+        """Partition the outputs of the Map.
+
+        The partition will only look at the direct intermediate outputs of the
+        Map. The outputs will be two sets, defined as:
+        - The independent outputs `\mathcal{I}`:
+            These are output nodes, whose output does not depend on the blocked
+            dimension. These nodes can be relocated between the outer and inner map.
+        - The dependent output `\mathcal{D}`:
+            These are the output nodes, whose output depend on the blocked dimension.
+            Thus they can not be relocated between the two maps, but will remain
+            inside the inner scope.
+
+        In case the function fails to compute the partition `None` is returned.
+
+        Args:
+            map_entry: The map entry node.
+            block_param: The Map variable that should be blocked.
+            state: The state on which we operate.
+            sdfg: The SDFG in which we operate on.
+
+        Note:
+            - Currently this function only considers the input Memlets and the
+                `used_symbol` properties of a Tasklet.
+            - Furthermore only the first level is inspected.
+        """
+        block_independent: set[dace_nodes.Node] = set()  # `\mathcal{I}`
+        block_dependent: set[dace_nodes.Node] = set()  # `\mathcal{D}`
+
+        # Find all nodes that are adjacent to the map entry.
+        nodes_to_partition: set[dace_nodes.Node] = {edge.dst for edge in state.out_edges(map_entry)}
+
+        # Now we examine every node and assign them to one of the sets.
+        #  Note that this is only tentative and we will later inspect the
+        #  outputs of the independent node and reevaluate their classification.
+        for node in nodes_to_partition:
```

Comment:
Same comment as above. The combination of `return` and `continue` statements in the same loop makes the code very hard to reason about. Please split it in smaller pieces or refactor the logic to avoid the combination of different loop flow statements

Reply:
I have now refactored the class.